pkg/apis/stable/v1alpha1/fleetautoscaler.go

```diff
@@ -102,37 +102,17 @@ type FleetAutoscalerStatus struct {
 	// lastScaleTime is the last time the FleetAutoscaler scaled the attached fleet,
 	// +optional
 	LastScaleTime *metav1.Time `json:"lastScaleTime"`
-
-	// AbleToScale indicates that we can access the target fleet
-	AbleToScale bool `json:"ableToScale"`
-
-	// ScalingLimited indicates that the calculated scale would be above or below the range
-	// defined by MinReplicas and MaxReplicas, and has thus been capped.
-	ScalingLimited bool `json:"scalingLimited"`
-}
-
-// ValidateUpdate validates when an update occurs
-func (fas *FleetAutoscaler) ValidateUpdate(new *FleetAutoscaler, causes []metav1.StatusCause) []metav1.StatusCause {
-	if fas.Spec.FleetName != new.Spec.FleetName {
-		causes = append(causes, metav1.StatusCause{
-			Type:    metav1.CauseTypeFieldValueInvalid,
-			Field:   "fleetName",
-			Message: "fleetName cannot be updated",
-		})
-	}
-
-	return new.ValidateAutoScalingSettings(causes)
 }
 
-//ValidateAutoScalingSettings validates the FleetAutoscaler scaling settings
-func (fas *FleetAutoscaler) ValidateAutoScalingSettings(causes []metav1.StatusCause) []metav1.StatusCause {
+//Validate validates the FleetAutoscaler scaling settings
+func (fas *FleetAutoscaler) Validate(causes []metav1.StatusCause) []metav1.StatusCause {
 	if fas.Spec.Policy.Type == BufferPolicyType {
 		causes = fas.Spec.Policy.Buffer.ValidateAutoScalingBufferPolicy(causes)
 	}
 	return causes
 }
 
-//ValidateAutoScalingSettings validates the FleetAutoscaler Buffer policy settings
+//Validate validates the FleetAutoscaler Buffer policy settings
 func (b *BufferPolicy) ValidateAutoScalingBufferPolicy(causes []metav1.StatusCause) []metav1.StatusCause {
```

Comment:
I think you meant to change the name of the function too.

Reply:
Actually no - but that comment shouldn't be changed. Weird the linter didn't pick that up?

Reply:
Fixed comment. Intellij got a little aggressive with an automatic refactor. Thanks for the pickup!

Reply:
Or actually, are you saying this should be something like `ValidateBufferPolicy` ? (That would make more sense)

Reply:
SGTM! Will make the change. Thanks for the quick response.